java/arcs/core/host/SdkHandleManager.kt

```diff
@@ -0,0 +1,432 @@
+package arcs.core.host
+
+import arcs.core.common.Id
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperationAtTime
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+import arcs.core.storage.Callbacks
+import arcs.core.storage.Handle as StorageHandle
+import arcs.core.storage.StorageKey
+import arcs.core.storage.api.Entity
+import arcs.core.storage.api.EntitySpec
+import arcs.core.storage.api.Handle
+import arcs.core.storage.api.ReadWriteCollection
+import arcs.core.storage.api.ReadWriteSingleton
+import arcs.core.storage.api.ReadableCollection
+import arcs.core.storage.api.ReadableSingleton
+import arcs.core.storage.api.WritableCollection
+import arcs.core.storage.api.WritableSingleton
+import arcs.core.storage.handle.CollectionImpl
+import arcs.core.storage.handle.HandleManager
+import arcs.core.storage.handle.SetData
+import arcs.core.storage.handle.SetHandle
+import arcs.core.storage.handle.SetOp
+import arcs.core.storage.handle.SingletonData
+import arcs.core.storage.handle.SingletonHandle
+import arcs.core.storage.handle.SingletonImpl
+import arcs.core.storage.handle.SingletonOp
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+typealias Sender = (block: suspend () -> Unit) -> Unit
+typealias SingletonSenderCallbackAdapter<E> =
+    SenderCallbackAdapter<SingletonData<RawEntity>, SingletonOp<RawEntity>, RawEntity?, E?>
+typealias CollectionSenderCallbackAdapter<E> =
+    SenderCallbackAdapter<SetData<RawEntity>, SetOp<RawEntity>, Set<RawEntity>, E>
+
+/**
+ * Wraps a [HandleManager] and creates SDK handles based on [HandleMode], such as
+ * [ReadableSingleton] for [HandleMode.Read]. To obtain a [HandleHolder], use
+ * `arcs_kt_schema` on a manifest file to generate a `{ParticleName}Handles' class, and
+ * invoke its default constructor, or obtain it from the [BaseParticle.handles] field.
+ */
+class SdkHandleManager(val handleManager: HandleManager) {
+    /**
+     * Create a [SingletonHandle] given a [StorageKey], [Schema],  a [HandleHolder] with
+     * [EntitySpec] definitions indexed by `handleName`. Populates the correct [HandleHolder]
+     * field with handle and returns newly created [SingletonHandle].
+     *
+     * @property handleHolder contains handle and entitySpec declarations
+     * @property handleName name for the handle, must be present in [HandleHolder.entitySpecs]
+     * @property storageKey a [StorageKey]
+     * @property schema the [Schema] for this [StorageKey]
+     * @property handleMode whether a handle is Read,Write, or ReadWrite (default)
+     * @property sender block used to execute callback lambdas
+     * @property idGenerator used to generate unique IDs for newly stored entities.
+     */
+    suspend fun sdkSingletonHandle(
```

Comment:
nit: unless this class provides ways to create both sdk and non-sdk handles I would think just `singletonHandle` or `createSingletonHandle` would be sufficient here (and similar for below methods), since `sdk` is implicit in the containing class name, and a verb prefix is often common (but I am a noob re: Kotlin idioms, so definitely defer to others).

Reply:
I tried to pattern this after the HandleManager which omits the 'create' prefix, but I can change it.

Reply:
Done